common/thorhelper/thorread.cpp

```diff
@@ -61,21 +212,18 @@ class DiskRowReader : extends CInterfaceOf<IAllocRowStream>, implements IRawRowS
 
 
 protected:
-    offset_t getLocalOffset();
+    virtual offset_t getLocalOffset();
 
 protected:
-    Owned<ISerialStream> input;
+    Owned<ISerialStream> inputStream;
     Owned<IFileIO> inputfileio;
-    CThorContiguousRowBuffer inputBuffer;
+    CThorContiguousRowBuffer inputBuffer;    // more: move to derived classes.
     Owned<IEngineRowAllocator> outputAllocator;
-    Owned<const IDynamicTransform> translator;
-    Owned<const IKeyTranslator> keyedTranslator;
-    Linked<IOutputMetaData> expectedDiskMeta = nullptr;
-    Linked<IOutputMetaData> projectedDiskMeta = nullptr;
-    Linked<IOutputMetaData> actualDiskMeta = nullptr;
-    unsigned expectedCrc = 0;
-    unsigned projectedCrc = 0;
-    unsigned actualCrc = 0;
+    RtlDynamicRowBuilder allocatedBuilder;
+    const IDynamicTransform * translator;
+    const IKeyTranslator * keyedTranslator;
```

Comment:
style: is there actual overhead if these are initialzed to null here? If not seems preferable for readability/future safety.

Reply:
yes, I think we should require all members are initialised - unless there is a very good reason not to